### ISSUE TITLE
How to get all users belong to a specific role?

### ISSUE THREAD

**Issue by haohello** [strongloop/loopback#332]
What I would like to accomplish is: get all users with the role of 'admin', is there an easier way to do this in loopback?

**Comment by mneil** [comment]
Seems there are several open issues with roles not correctly being included with users. The way I ended up making this work was:

`server/boot/role-mapping.js`
```
module.exports = function(app) {

  var RoleMapping = app.models.RoleMapping;
  /**
   * Get user ID's by role name
   * @param role
   * @param callback
   */
  RoleMapping.usersIDByRole = function(role, callback){

    RoleMapping.app.models.Role.byName(role, function(err, role){

      if( err ) return callback(err);

      RoleMapping.find({
        where: {
          roleId: role.id,
          principalType: RoleMapping.USER
        }
      }, function(err, mappings){

        if( err ) return callback(err);

        var users = mappings.map(function (m) {
          return m.principalId;
        });

        callback(null, users);
      });

    });

  }

};
```

Then in my users model `common/models/user.js`
```
/**
   * Return the first client ROLE user that belongs to an account
   * @param account
   * @param callback
   */
  User.getUsersByRole = function(role, callback) {

    User.app.models.RoleMapping.usersIDByRole(role, function(err, users) {

      User.findOne({
        where : {
          _id : {inq : users}
        }
      }, callback);
    });
  };
```

**Comment by hilarudeensa** [comment]
@mneil Thanks, your solution is working fine. However in `RoleMapping.app.models.Role.byName`, the function `byName` is saying undefined. Hence, I have modified the above portion as follows
 `RoleMapping.app.models.Role.findOne({where: {name:role}}, function(err, role){
 // ...
});

**Comment by mneil** [comment]
Ah yes, sorry @hilarudeensa I didn't define that function in my example.

In `./server/boot/role.js`

```
module.exports = function(app) {

  var Role = app.models.Role;

  /**
   * Get a role by name
   * @param name
   */
  Role.byName = function(name, callback){

    Role.findOne({where: {name: name}}, callback);
  }
}
```

**Comment by superkhau** [comment]
Are you guys still running into issues? Can I close this?

**Comment by mneil** [comment]
I think you should close this issue. It's related to https://github.com/strongloop/loopback/issues/1441. I was just suggesting a workaround that I used. A couple of the suggestions about coercing the id to a mongo ObjectId in the issue I just linked to really is a better solution though.

**Comment by superkhau** [comment]
@mneil Thanks for the response. Closing.

Please open a new issue if you guys are still running into problems.

**Comment by codyolsen** [comment]
Please note that the `adminRole.users` method used in @raymondfeng's solution is currently broken. I've submitted a simple fix in PR #2669.
